### DISCUSSION TITLE
Mine- and Role-permissions from V8

### DISCUSSION THREAD

**Discussion by JLKM** [directus/directus#10750] [Archive]
Access for users to update their own (self-created) items/records is crucial for the system, I wan't to build in Directus. This feature looks a lot like the `mine`- and `role`-permissions found in [version 8](https://v8.docs.directus.io/guides/permissions.html#collection-level).

I haven't been able to find anything similar in V9. Have made an attempt on filter rules in custom role settings, but no luck. Being able to set a filter rule like `user_created == me/owner`, would push V9 up to V8 speed in this matter.

Really looking forward to that feature being implemented in V9 (or have I missed something).

**Comment by licitdev** [comment]
This feature has been implemented in https://github.com/directus/directus/pull/9376.

You are able to use the following filter in permissions.
`user_created` equals `$CURRENT_USER.id`

For more details, please refer to https://docs.directus.io/configuration/filter-rules/#dynamic-variables

**Reply by JLKM** [reply]
Thank you for the input. Unfortunately this configuration with full permission to read and create doesn't allow the user to update her own record. Any suggestions for fine tuning?
![billede](https://user-images.githubusercontent.com/18215374/147670659-cd288e1b-969c-44da-af1c-6dfafcdd81f0.png)

**Reply by licitdev** [reply]
Your permission seems to be correct.
Have you selected fields allowed to be updated?

<img width="148" alt="image" src="https://user-images.githubusercontent.com/26413686/147671222-65661322-d933-4bc2-b8e9-044d261f8976.png">

**Comment by JLKM** [comment]
My bad. You're absolutely right, and it works. You just have to get used to my author-role being able to actually edit records created by another user. The author is not allowed to save the edited record, however. On the other side the author is able to save a copy of a record created by another user. This is spot - and very useful. Thank you, @licitdev, for your guidance :-)
